recipe/conda_build_config.yaml

```
@@ -6,3 +6,5 @@ libevent:
   - "2.1.10"
 lmdb:
   - "0.9.22"
+folly:
+  - "2023.09.25.00"	
```

Can you indicate the reason for pinning this specific version? For instance:
```suggestion
folly:
  # This version of folly makes this builds usable into more environment.
  - "2023.09.25.00"
```

Also, I think we can also pin folly to `2023.09.25.00` on `main` and on the same version for `1.6.X` as its last builds as well since:
 - rolling releases of folly do not come with stability (_e.g._ [folly recently opt to use non portable macro of Apple Clang](https://quantstack.slack.com/archives/C0553TEJ63F/p1700156532900529)).
 - an [old version of folly is also pinned for vcpkg](https://github.com/man-group/ArcticDB/blob/master/cpp/vcpkg.json#L148), and we could upgrade to pin to reduce discrepancy.
 - on conda-forge, `folly` only is a dependency of `arcticdb` and `pyhdk`

This will constraint environements with specific version of the dependencies of folly though.

What do you think?

---

Let's test whether the change on this branch helps the situation first, and then I agree we can update main too if it does